packages/api/src/controllers/stream.ts

```diff
@@ -1097,10 +1123,8 @@ async function publishDelayedRecordingReadyHook(
       userId: session.userId,
       sessionId: session.id,
       payload: {
-        recordingUrl: await getRecordingUrl(ingest, session),
-        mp4Url: await getRecordingUrl(ingest, session, true),
```

Comment:
Can you add those from the cannon instead so we don't break compatibility?

Reply:
Isn't `cannon` only for new recordings, which will always work with Recording V2? Not sure I understand why we need backward-compatibility for this part.

Reply:
Oh, the cannon is not used only for recordings. The recordings logic there is actually a "plugin" on top of its main responsibility: firing webhooks. So this is the `recording.ready` webhook, which fires 1 minute after the stream is done, which can let a user's backend know that they can do sth with the recording. 

Ideally, we would also change these webhooks to only call the `recording.ready` one the recording asset is created, since that's how users will really interact with it. I think we can do a low effort now, by keeping the `recording.ready` when the asset is _created_ (started processing) and then tell users to wait for the proper VOD webhooks to know when it's actually available.

So this `payload` here is actually the **webhook payload**. I was thinking that we could omit the URLs here, since they're not ready yet, but add them from that `recording.ready` special handling on the webhook cannon. Maybe also an `assetId` field? Theoretically the URL won't work immediately when we send the webhook, but I think it's still better than removing the field that some users might be depending on. WDYT?